lib/Doctrine/DBAL/Driver/SQLite3/SQLite3Connection.php

```diff
@@ -0,0 +1,216 @@
+<?php
+/*
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license. For more information, see
+ * <http://www.doctrine-project.org>.
+ */
+
+namespace Doctrine\DBAL\Driver\SQLite3;
+
+use Doctrine\DBAL\Driver\Connection;
+use Doctrine\DBAL\Driver\ServerInfoAwareConnection;
+
+/**
+ * SQLite3 implementation of the Connection interface.
+ *
+ * @since 2.6
+ * @author Ben Morel <benjamin.morel@gmail.com>
+ * @author Bill Schaller <bill@zeroedin.com>
+ */
+class SQLite3Connection extends SQLite3Abstract implements Connection, ServerInfoAwareConnection
+{
+    /**
+     * @param string      $filename
+     * @param int|null    $flags
+     * @param string|null $encryption_key
+     *
+     * @throws SQLite3Exception
+     */
+    public function __construct($filename, $flags = null, $encryption_key = null)
+    {
+        try {
+            if ($flags === null) {
+                $this->sqlite3 = new \SQLite3($filename);
+            } elseif ($encryption_key === null) {
+                $this->sqlite3 = new \SQLite3($filename, $flags);
+            } else {
+                $this->sqlite3 = new \SQLite3($filename, $flags, $encryption_key);
+            }
+
+            $this->sqlite3->enableExceptions(true);
+        } catch (\Exception $e) {
+            throw $this->wrapDriverException($e);
+        }
+    }
+
+    /**
+     * Returns the underlying SQLite3 object.
+     *
+     * @return \SQLite3
+     */
+    public function getSQLite3()
```

Comment:
The only other driver that allows this is mysqli. I'm really strongly against returning the SQLite3 instance as well. If we want to add specific support methods for SQLite3 features, I'm just fine with that.

If you have external code that uses SQLite3 directly, we could simply change the driver so that it extends \SQLite3 instead of completely wrapping it in a non-compatible interface. Methods we don't want callable by userland (such as enableExceptions) we could simply add a stub that throws something like a NotImplementedException.